src/Symfony/Component/HttpClient/Response/AsyncContext.php

```diff
@@ -152,13 +153,18 @@ public function getResponse(): ResponseInterface
      */
     public function replaceRequest(string $method, string $url, array $options = []): ResponseInterface
     {
-        $this->info['previous_info'][] = $this->response->getInfo();
+        $this->info['previous_info'][] = $info = $this->response->getInfo();
         if (null !== $onProgress = $options['on_progress'] ?? null) {
             $thisInfo = &$this->info;
             $options['on_progress'] = static function (int $dlNow, int $dlSize, array $info) use (&$thisInfo, $onProgress) {
                 $onProgress($dlNow, $dlSize, $thisInfo + $info);
             };
         }
+        if (0 < ($info['max_duration'] ?? 0) && 0 < ($info['total_time'] ?? 0) && !isset($options['max_duration'])) {
```

Comment:
I'm not sure if I got what you did here, but unless I remove `!` from `!isset($options['max_duration'])` I couldn't make `max_duration` option work as expected when testing it. I will leave more details at the PR's comment area.

Reply:
I actually removed the check for `$options['max_duration']`

Reply:
is the added test realistic/good enough then? I stared a while at this check but was (wrongly?) reassured by the test reaching the expected exception.